### ISSUE TITLE
Save empty files when using SpatialDimPlot and SpatialFeaturePlot 

### ISSUE THREAD

**Issue by JJM-labs** [satijalab/seurat#9432]
When I use `SpatialFeaturePlot`, the results showed in Rstudio are correct (fig.1)：
`SpatialFeaturePlot(data, features = "TP53")`

But, when I save as a pdf file the cell point not included (fig.2)：
`ggsave("Spatial_TP53.pdf", w=5, h=5)`

Furthermore, when I set the parameter: `image.alpha = 0`, the saved image will be empty (fig.3).

```
R version 4.3.3 (2024-02-29 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 11 x64 (build 22631)

Matrix products: default

locale:
[1] LC_COLLATE=Chinese (Simplified)_China.utf8  LC_CTYPE=Chinese (Simplified)_China.utf8   
[3] LC_MONETARY=Chinese (Simplified)_China.utf8 LC_NUMERIC=C                               
[5] LC_TIME=Chinese (Simplified)_China.utf8    

time zone: Asia/Shanghai
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] patchwork_1.3.0    ggplot2_3.5.1      Seurat_5.1.0       SeuratObject_5.0.2 sp_2.1-4          

loaded via a namespace (and not attached):
  [1] deldir_2.0-4           pbapply_1.7-2          gridExtra_2.3          rlang_1.1.4            magrittr_2.0.3        
  [6] RcppAnnoy_0.0.22       matrixStats_1.4.1      ggridges_0.5.6         compiler_4.3.3         spatstat.geom_3.3-3   
 [11] png_0.1-8              vctrs_0.6.5            reshape2_1.4.4         stringr_1.5.1          pkgconfig_2.0.3       
 [16] fastmap_1.2.0          utf8_1.2.4             promises_1.3.0         purrr_1.0.2            jsonlite_1.8.9        
 [21] goftest_1.2-3          later_1.3.2            spatstat.utils_3.1-0   irlba_2.3.5.1          parallel_4.3.3        
 [26] cluster_2.1.6          R6_2.5.1               ica_1.0-3              stringi_1.8.4          RColorBrewer_1.1-3    
 [31] spatstat.data_3.1-2    reticulate_1.39.0      parallelly_1.38.0      spatstat.univar_3.0-1  lmtest_0.9-40         
 [36] scattermore_1.2        Rcpp_1.0.13            tensor_1.5             future.apply_1.11.2    zoo_1.8-12            
 [41] sctransform_0.4.1      httpuv_1.6.15          Matrix_1.6-5           splines_4.3.3          igraph_2.1.1          
 [46] tidyselect_1.2.1       rstudioapi_0.17.1      abind_1.4-8            spatstat.random_3.3-2  codetools_0.2-19      
 [51] miniUI_0.1.1.1         spatstat.explore_3.3-3 listenv_0.9.1          lattice_0.22-5         tibble_3.2.1          
 [56] plyr_1.8.9             withr_3.0.1            shiny_1.9.1            ROCR_1.0-11            Rtsne_0.17            
 [61] future_1.34.0          fastDummies_1.7.4      survival_3.5-8         polyclip_1.10-7        fitdistrplus_1.2-1    
 [66] pillar_1.9.0           KernSmooth_2.23-22     plotly_4.10.4          generics_0.1.3         RcppHNSW_0.6.0        
 [71] munsell_0.5.1          scales_1.3.0           globals_0.16.3         xtable_1.8-4           glue_1.8.0            
 [76] lazyeval_0.2.2         tools_4.3.3            data.table_1.16.2      RSpectra_0.16-2        RANN_2.6.2            
 [81] leiden_0.4.3.1         dotCall64_1.2          cowplot_1.1.3          grid_4.3.3             tidyr_1.3.1           
 [86] colorspace_2.1-1       nlme_3.1-164           cli_3.6.3              spatstat.sparse_3.1-0  spam_2.11-0           
 [91] fansi_1.0.6            viridisLite_0.4.2      dplyr_1.1.4            uwot_0.2.2             gtable_0.3.6          
 [96] digest_0.6.37          progressr_0.14.0       ggrepel_0.9.6          htmlwidgets_1.6.4      farver_2.1.2          
[101] htmltools_0.5.8.1      lifecycle_1.0.4        httr_1.4.7             mime_0.12              MASS_7.3-60.0.1 

```
fig.1
![QQ20241027-221329](https://github.com/user-attachments/assets/3f53ea6d-d90f-406d-86f9-bbd8f362f2e5)

fig.2
![QQ20241027-221827](https://github.com/user-attachments/assets/f57644a6-119c-4384-bea4-708f9af6fa97)

fig.3
![QQ20241027-222419](https://github.com/user-attachments/assets/7b4be50c-4cbf-4a8a-98ee-53bff4a72da8)

**Comment by JJM-labs** [comment]
> 当我使用 时，Rstudio 中显示的结果是正确的（图 1）：`SpatialFeaturePlot` `SpatialFeaturePlot(data, features = "TP53")`
> 
> 但是，当我另存为 pdf 文件时，单元格点不包括在内（图 2）： `ggsave("Spatial_TP53.pdf", w=5, h=5)`
> 
> 此外，当我设置参数 ： 时，保存的图像将是空的（图 3）。`image.alpha = 0`
> 
> ```
> R version 4.3.3 (2024-02-29 ucrt)
> Platform: x86_64-w64-mingw32/x64 (64-bit)
> Running under: Windows 11 x64 (build 22631)
> 
> Matrix products: default
> 
> locale:
> [1] LC_COLLATE=Chinese (Simplified)_China.utf8  LC_CTYPE=Chinese (Simplified)_China.utf8   
> [3] LC_MONETARY=Chinese (Simplified)_China.utf8 LC_NUMERIC=C                               
> [5] LC_TIME=Chinese (Simplified)_China.utf8    
> 
> time zone: Asia/Shanghai
> tzcode source: internal
> 
> attached base packages:
> [1] stats     graphics  grDevices utils     datasets  methods   base     
> 
> other attached packages:
> [1] patchwork_1.3.0    ggplot2_3.5.1      Seurat_5.1.0       SeuratObject_5.0.2 sp_2.1-4          
> 
> loaded via a namespace (and not attached):
>   [1] deldir_2.0-4           pbapply_1.7-2          gridExtra_2.3          rlang_1.1.4            magrittr_2.0.3        
>   [6] RcppAnnoy_0.0.22       matrixStats_1.4.1      ggridges_0.5.6         compiler_4.3.3         spatstat.geom_3.3-3   
>  [11] png_0.1-8              vctrs_0.6.5            reshape2_1.4.4         stringr_1.5.1          pkgconfig_2.0.3       
>  [16] fastmap_1.2.0          utf8_1.2.4             promises_1.3.0         purrr_1.0.2            jsonlite_1.8.9        
>  [21] goftest_1.2-3          later_1.3.2            spatstat.utils_3.1-0   irlba_2.3.5.1          parallel_4.3.3        
>  [26] cluster_2.1.6          R6_2.5.1               ica_1.0-3              stringi_1.8.4          RColorBrewer_1.1-3    
>  [31] spatstat.data_3.1-2    reticulate_1.39.0      parallelly_1.38.0      spatstat.univar_3.0-1  lmtest_0.9-40         
>  [36] scattermore_1.2        Rcpp_1.0.13            tensor_1.5             future.apply_1.11.2    zoo_1.8-12            
>  [41] sctransform_0.4.1      httpuv_1.6.15          Matrix_1.6-5           splines_4.3.3          igraph_2.1.1          
>  [46] tidyselect_1.2.1       rstudioapi_0.17.1      abind_1.4-8            spatstat.random_3.3-2  codetools_0.2-19      
>  [51] miniUI_0.1.1.1         spatstat.explore_3.3-3 listenv_0.9.1          lattice_0.22-5         tibble_3.2.1          
>  [56] plyr_1.8.9             withr_3.0.1            shiny_1.9.1            ROCR_1.0-11            Rtsne_0.17            
>  [61] future_1.34.0          fastDummies_1.7.4      survival_3.5-8         polyclip_1.10-7        fitdistrplus_1.2-1    
>  [66] pillar_1.9.0           KernSmooth_2.23-22     plotly_4.10.4          generics_0.1.3         RcppHNSW_0.6.0        
>  [71] munsell_0.5.1          scales_1.3.0           globals_0.16.3         xtable_1.8-4           glue_1.8.0            
>  [76] lazyeval_0.2.2         tools_4.3.3            data.table_1.16.2      RSpectra_0.16-2        RANN_2.6.2            
>  [81] leiden_0.4.3.1         dotCall64_1.2          cowplot_1.1.3          grid_4.3.3             tidyr_1.3.1           
>  [86] colorspace_2.1-1       nlme_3.1-164           cli_3.6.3              spatstat.sparse_3.1-0  spam_2.11-0           
>  [91] fansi_1.0.6            viridisLite_0.4.2      dplyr_1.1.4            uwot_0.2.2             gtable_0.3.6          
>  [96] digest_0.6.37          progressr_0.14.0       ggrepel_0.9.6          htmlwidgets_1.6.4      farver_2.1.2          
> [101] htmltools_0.5.8.1      lifecycle_1.0.4        httr_1.4.7             mime_0.12              MASS_7.3-60.0.1 
> ```
> 
> 图 1 ![QQ20241027-221329](https://private-user-images.githubusercontent.com/161120359/380503981-3f53ea6d-d90f-406d-86f9-bbd8f362f2e5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzAwNDI2NzMsIm5iZiI6MTczMDA0MjM3MywicGF0aCI6Ii8xNjExMjAzNTkvMzgwNTAzOTgxLTNmNTNlYTZkLWQ5MGYtNDA2ZC04NmY5LWJiZDhmMzYyZjJlNS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDI3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAyN1QxNTE5MzNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1lNTVkYmQyOGM3MjgzZmM1ZjA1MzQ1M2Q3ZDdiYjBhYTU0ZDJhMzQ2NTI4ZDk1MzYxYjYxMDgyZjdlNmY5MTUwJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.1CJVH49uvJSWMev3u20MyGkcqAIxHvcHuL8rcRaTvs8)
> 
> 图 2 ![QQ20241027-221827](https://private-user-images.githubusercontent.com/161120359/380504197-f57644a6-119c-4384-bea4-708f9af6fa97.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzAwNDI2NzMsIm5iZiI6MTczMDA0MjM3MywicGF0aCI6Ii8xNjExMjAzNTkvMzgwNTA0MTk3LWY1NzY0NGE2LTExOWMtNDM4NC1iZWE0LTcwOGY5YWY2ZmE5Ny5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDI3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAyN1QxNTE5MzNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05YmZjMTcwMzE5MGE5NzgwOTI0YWJlOGZmODgwMWE1MmQ1ODRhNzcyOTdkNjU5NTdhNTVkMDI2YTUxMjA1M2NiJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.EOrRIMCPLRr7V0pFAdzlvAVIG6KWaGBdYo9mRZUYO5o)
> 
> 图 3 ![QQ20241027-222419](https://private-user-images.githubusercontent.com/161120359/380504655-7b4be50c-4cbf-4a8a-98ee-53bff4a72da8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzAwNDI2NzMsIm5iZiI6MTczMDA0MjM3MywicGF0aCI6Ii8xNjExMjAzNTkvMzgwNTA0NjU1LTdiNGJlNTBjLTRjYmYtNGE4YS05OGVlLTUzYmZmNGE3MmRhOC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDI3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAyN1QxNTE5MzNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1kN2JjODBmY2U5NzI3OGUzMGQ5OTI4ODVhZDQ4ZmQzYTg4NDFiN2MyNmJlNDAyMGUzNGM4MmI2ZmMwM2FlMWE1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.fH6SZw0mt77G7pkAUi7Vp6RSFkRBhddcElpCIApQ55o)

**Comment by JJM-labs** [comment]
I have solved this problem, download `seurat 5.0.3`，not `5.1.0`.
This could be a version compatibility issue, but I can't determine which packages are conflicting.

**Comment by samuel-marsh** [comment]
Hi,

Not member of dev team but hopefully can be helpful.  I cannot replicate your issue with 5.1.0 using brain data from SeuratData.  If you want to use 5.1.0 I would suggest trying a fresh install of 5.1.0 and making sure your ggplot2 version is up to date as well.  If this continues to be issue please let me know here and I'll reopen the issue.

Best,
Sam
